### PR TITLE
feat(driving-license): start on advanced driving-license

### PR DESCRIPTION
**Pull request by stjanilofts**
Only adding the first step for now until we have the correct endpoints to use.

Behind a feature flag "isDrivingLicenseAdvancedEnabled"

## Release Notes

- **New Features**
  - Introduced the `AdvancedLicense` component for selecting advanced driving licenses.
  - Added a new section for advanced license selection in the application form, enhancing user interactivity.
  - Implemented new constants and enumerations for advanced license types and requirements.
  - Added multiple new message definitions for various advanced driving license categories.

- **Improvements**
  - Enhanced form configurability with the addition of an `allowAdvanced` parameter for advanced license options.
  - Improved localization capabilities with new message entries for advanced license descriptions and requirements.

- **Bug Fixes**
  - Removed unnecessary debug output from the `Alert` component to streamline functionality.

### CODE REVIEW

File: libs/application/templates/driving-license/src/forms/prerequisites/sectionAdvancedLicenseSelection.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    buildMultiField({
+      id: 'advancedLicenseSelectionFields',
+      title: m.applicationForAdvancedLicenseTitle,
+      description: m.applicationForAdvancedLicenseDescription,
+      children: [
+        buildCustomField({
+          id: 'advancedLicense',
+          title: 'Advanced License',
+          component: 'AdvancedLicense',
+        }),
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/forms/prerequisites/sectionAdvancedLicenseSelection.ts:31]: this is not applicable

File: libs/application/templates/driving-license/src/lib/dataSchema.ts [hunk 2]
```diff
@@ -36,7 +45,10 @@ export const dataSchema = z.object({
   ]),
   requirementsMet: z.boolean().refine((v) => v),
   certificate: z.array(z.enum([YES, NO])).nonempty(),
-  applicationFor: z.enum([B_FULL, B_TEMP, BE, B_FULL_RENEWAL_65]),
+  applicationFor: z.enum([B_FULL, B_TEMP, BE, B_FULL_RENEWAL_65, B_ADVANCED]),
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/dataSchema.ts:48]: doesn't seem possible in this library, not this version at least

File: libs/application/templates/driving-license/src/forms/prerequisites/sectionApplicationFor.ts [hunk 3]
```diff
@@ -112,6 +114,17 @@ export const sectionApplicationFor = (
...
               }
 
+              if (allowAdvanced) {
+                options = options.concat({
+                  label: m.applicationForAdvancedLicenseTitle,
+                  subLabel: m.applicationForAdvancedLicenseDescription,
+                  value: B_ADVANCED,
+                  disabled: !categories?.some(
+                    (c) => c.nr.toUpperCase() === 'B' && c.validToCode !== 8,
+                  ),
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/forms/prerequisites/sectionApplicationFor.ts:124]: I would rather do this in a refactor PR for the whole driving-license project

File: libs/application/templates/driving-license/src/lib/messages.ts [hunk 4]
```diff
@@ -878,6 +878,121 @@ export const m = defineMessages({
...
+  applicationForAdvancedLicenseLabelD1: {
+    id: 'dl.application:applicationForAdvancedLicenseLabelD1',
+    defaultMessage: 'D1 lýsing',
+    description: 'D1 description',
+  },
+  applicationForAdvancedLicenseLabelD1A: {
+    id: 'dl.application:applicationForAdvancedLicenseLabelD1A',
+    defaultMessage: 'D1A lýsing',
+    description: 'D1A description',
+  },
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/messages.ts:925]: I will get the correct messages later

File: libs/application/templates/driving-license/src/lib/messages.ts [hunk 5]
```diff
@@ -878,6 +878,121 @@ export const m = defineMessages({
...
+  applicationForAdvancedLicenseTitleDE: {
+    id: 'dl.application:applicationForAdvancedLicenseTitleDE',
+    defaultMessage: 'DE titill',
+    description: 'DE title',
+  },
+  applicationForAdvancedLicenseLabelDE: {
+    id: 'dl.application:applicationForAdvancedLicenseLabelDE',
+    defaultMessage: 'DE lýsing',
+    description: 'DE description',
+  },
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/messages.ts:995]: I will get the correct messages later

File: libs/application/templates/driving-license/src/lib/messages.ts [hunk 6]
```diff
@@ -878,6 +878,121 @@ export const m = defineMessages({
...
+  applicationForAdvancedLicenseLabelD: {
+    id: 'dl.application:applicationForAdvancedLicenseLabelD',
+    defaultMessage: 'D lýsing',
+    description: 'D description',
+  },
+  applicationForAdvancedLicenseLabelDA: {
+    id: 'dl.application:applicationForAdvancedLicenseLabelDA',
+    defaultMessage: 'DA lýsing',
+    description: 'DA description',
+  },
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/messages.ts:955]: I will get the correct messages later

File: libs/application/templates/driving-license/src/lib/constants.ts [hunk 7]
```diff
@@ -8,9 +8,127 @@ export enum ApiActions {
 export const B_FULL = 'B-full'
 export const B_TEMP = 'B-temp'
 export const B_FULL_RENEWAL_65 = 'B-full-renewal-65'
+export const B_ADVANCED = 'B-advanced'
```
[the remark this answers is not in the document]
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/constants.ts:11]: I don't know what charge item code to use at this stage

File: libs/application/templates/driving-license/src/lib/constants.ts [hunk 8]
```diff
@@ -8,9 +8,127 @@ export enum ApiActions {
...
+  minAge: number
+  group: keyof typeof MainAdvancedLicense
+  code: keyof typeof MainAdvancedLicense
+  professional?: {
+    minAge: number
+    code: keyof typeof ProfessionalAdvancedLicense
+  }
+}
+
+export const advancedLicenseMap: AdvancedLicenseMapItem[] = [
```
Comment (albinagu) [inline libs/application/templates/driving-license/src/lib/constants.ts:61]: verður þetta ekki sótt?

**Review by albinagu — APPROVED** [review]

File: libs/application/templates/driving-license/src/lib/constants.ts (hunk 8 above)
Comment (stjanilofts) [reply libs/application/templates/driving-license/src/lib/constants.ts:61]: held ekki, ekki núna allavega en gæti verið á einhverjum tímapunkti 🤷

**Pull request merged**